### PR TITLE
Add logging for Linux AppService Containers

### PR DESCRIPTION
**Pull request by bala16**
When running on appservice containers, use the mounted log folder to write logs. To avoid using up too much space on the worker, support rolling of log files.

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceEventGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            string scriptType, bool isDisabled)
+        {
+            var logger = _loggerFactory.GetOrCreate(FunctionsDetailsCategory);
+            WriteEvent(logger, $"{siteName},{functionName},{NormalizeString(inputBindings)},{NormalizeString(outputBindings)},{scriptType},{(isDisabled ? 1 : 0)}");
+        }
+
+        public void LogFunctionExecutionAggregateEvent(string siteName, string functionName, long executionTimeInMs,
+            long functionStartedCount, long functionCompletedCount, long functionFailedCount)
+        {
+            throw new NotImplementedException();
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceEventGenerator.cs]: The execution pipeline will be calling into this and the one below - won't these exceptions cause things to blow up?

File: src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceFileLogger.cs [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            _cancellationTokenSource = new CancellationTokenSource();
+
+            if (startOnCreate)
+            {
+                Start();
+            }
+        }
+
+        // Maximum number of files
+        public int MaxFileCount { get; set; } = 3;
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceFileLogger.cs]: Note we already have a FileWriter that we use for our other file system logs. It does auto purge, throttling, etc. Have you seen that? Would be good for us to not have to maintain 2 of these.

File: src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceEventGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    internal class LinuxAppServiceEventGenerator : LinuxEventGenerator, IEventGenerator
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceEventGenerator.cs]: Shouldn't LinuxEventGenerator inherit IEventGenerator, so all subclasses implement it?

File: src/WebJobs.Script.WebHost/Diagnostics/ILinuxAppServiceFileSystem.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    public interface ILinuxAppServiceFileSystem
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Diagnostics/ILinuxAppServiceFileSystem.cs]: You're adding this to facilitate mock tests? We already have a pattern for that. You can just use IFileSystem. Look for examples of Mock<IFileSystem> in the tests. I'd rather not have this new type.

File: src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceFileLogger.cs (hunk 2 above)
Comment (bala16) [reply src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceFileLogger.cs]: I saw that. but that class would need some changes to the filename format and few other things. so i thought it was better to keep this separate.

File: src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceEventGenerator.cs (hunk 3 above)
Comment (bala16) [reply src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceEventGenerator.cs]: fixed.

File: src/WebJobs.Script.WebHost/Diagnostics/ILinuxAppServiceFileSystem.cs (hunk 4 above)
Comment (bala16) [reply src/WebJobs.Script.WebHost/Diagnostics/ILinuxAppServiceFileSystem.cs]: Fixed.

File: src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceEventGenerator.cs (hunk 1 above)
Comment (bala16) [reply src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceEventGenerator.cs]: Fixed.

File: src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceFileLogger.cs [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+                // ignored
+            }
+        }
+
+        public virtual async Task ProcessLogQueue(object state)
+        {
+            while (!_cancellationTokenSource.IsCancellationRequested)
+            {
+                await InternalProcessLogQueue();
+                await Task.Delay(TimeSpan.FromSeconds(FlushFrequencySeconds));
```
[the remark this answers is not in the document]
Comment (bala16) [reply src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceFileLogger.cs]: Fixed.

File: src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceFileSystem.cs [hunk 6]
```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    public class LinuxAppServiceFileSystem : ILinuxAppServiceFileSystem
```
[the remark this answers is not in the document]
Comment (bala16) [reply src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceFileSystem.cs]: Removed this class.

**Pull request merged**